pyquil/api/_base_connection.py

```diff
@@ -33,127 +37,102 @@
-        :param dict json: JSON.
-        :return: A non-error response.
-        """
-        headers = {
-            'X-Api-Key': self.api_key,
-            'X-User-Id': self.user_id,
-            'Content-Type': 'application/json; charset=utf-8'
-        }
-        res = self._session.post(url, json=json, headers=headers)
-
-        # Print some nice info for unauthorized/permission errors.
-        if res.status_code == 401 or res.status_code == 403:
-            print("! ERROR:\n"
-                  "!   There was an issue validating your forest account.\n"
-                  "!   Have you run the pyquil-config-setup command yet?\n"
-                  "! The server came back with the following information:\n"
-                  "%s\n%s\n%s" % ("=" * 80, res.text, "=" * 80), file=sys.stderr)
-            print("! If you suspect this to be a bug in pyQuil or Rigetti Forest,\n"
-                  "! then please describe the problem in a GitHub issue at:\n!\n"
-                  "!      https://github.com/rigetticomputing/pyquil/issues\n", file=sys.stderr)
-
-        # Print some nice info for invalid input or internal server errors.
-        if res.status_code == 400 or res.status_code >= 500:
-            print("! ERROR:\n"
-                  "!   Server caught an error. This could be due to a bug in the server\n"
-                  "!   or a bug in your code. The server came back with the following\n"
-                  "!   information:\n"
-                  "%s\n%s\n%s" % ("=" * 80, res.text, "=" * 80), file=sys.stderr)
-            print("! If you suspect this to be a bug in pyQuil or Rigetti Forest,\n"
-                  "! then please describe the problem in a GitHub issue at:\n!\n"
-                  "!      https://github.com/rigetticomputing/pyquil/issues\n", file=sys.stderr)
-
-        res.raise_for_status()
-        return res
-
-    def _get_json(self, url):
-        """
-        Get JSON from a Forest endpoint.
-
-        :param str url: The full url to fetch
-        :return: Response object
-        """
-        headers = {
-            'X-Api-Key': self.api_key,
-            'X-User-Id': self.user_id,
-            'Content-Type': 'application/json; charset=utf-8'
-        }
-        return requests.get(url, headers=headers)
+def wait_for_job(get_job_fn, ping_time=None, status_time=None):
+    """
+    Wait for job logic
+    """
+    count = 0
+    while True:
+        job = get_job_fn()
+        if job.is_done():
+            break
+
+        if status_time and count % int(status_time / ping_time) == 0:
+            if job.is_queued():
+                print("job {} is currently queued at position {}".format(job.job_id, job.position_in_queue()))
+            elif job.is_running():
+                print("job {} is currently running".format(job.job_id))
+
+        time.sleep(ping_time)
+        count += 1
+
+    return job
+
+
+def get_json(session, url):
+    """
+    Get JSON from a Forest endpoint.
+    """
```

Comment:
based on what criteria do you add `:param ...:` and `:return:` statements? This function and the following two lack them

Reply:
For internal methods (this class is private) I'm a bit more laid-back. So yes I intentionally did not add them.